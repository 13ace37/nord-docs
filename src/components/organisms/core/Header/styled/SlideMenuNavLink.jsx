/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React from "react";
import posed from "react-pose";

import NavLink from "./NavLink";
import { slideMenuNavigationLinkPoseConfig } from "./poseConfig";

const Link = React.forwardRef((props, ref) => <NavLink ref={ref} {...props} />);

/**
 * An animated slide menu navigation link.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */
const SlideMenuNavLink = posed(Link)(slideMenuNavigationLinkPoseConfig);

export default SlideMenuNavLink;
