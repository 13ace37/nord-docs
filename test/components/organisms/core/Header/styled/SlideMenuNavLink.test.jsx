/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React from "react";

import { renderWithTheme } from "nord-docs-test-utils";
import SlideMenuNavLink from "organisms/core/Header/styled/SlideMenuNavLink";

describe("theme styles", () => {
  test("matches the snapshot", () => {
    const { container } = renderWithTheme(<SlideMenuNavLink>Nord</SlideMenuNavLink>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
