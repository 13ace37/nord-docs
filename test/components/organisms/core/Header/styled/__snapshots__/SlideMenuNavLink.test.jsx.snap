// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme styles matches the snapshot 1`] = `
.c1 {
  color: inherit;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:active,
.c1:focus,
.c1:hover,
.c1:visited {
  outline: none;
}

.c0 {
  border-radius: 0.25em;
  padding: 0.15em 0.5em;
  -webkit-transition: background-color 100ms ease-in-out;
  transition: background-color 100ms ease-in-out;
}

.c0:active,
.c0:focus,
.c0:hover {
  background-color: rgba(236,239,244,0.4);
}

<a
  class="c0 c1"
  href=""
  style="transform: none;"
>
  Nord
</a>
`;
